graphium/trainer/losses.py

```diff
@@ -64,7 +64,7 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
         # we apply softmax on the raw logits first
         softmax_input = self.softmax(input)
         # [batch_size, n_classes] * [n_classes] ([0, 1, 2...n_brakets-1]) -> [batch_size]
-        regression_input = torch.inner(softmax_input, self.brackets.to(input.device))
+        regression_input = torch.inner(softmax_input.float(), self.brackets.to(input.device))
```

Comment:
Required for mixed-precision (otherwise dtypes do not match in `torch.inner`.

Reply:
This could break things. you could perhaps do this?
```suggestion
        regression_input = torch.inner(softmax_input.to(self.brackets.dtype), self.brackets.to(input.device))
```